crates/autopilot/src/domain/auction/mod.rs

```diff
@@ -7,7 +7,7 @@ pub mod order;
 
 /// Replicates [`crate::model::Auction`].
 #[derive(Clone, Debug, PartialEq)]
-pub struct Auction {
+pub struct AuctionWithoutId {
```

Comment:
Maybe we could give this a better name (`RawAuctionData` perhaps?) or at least some doc comments somewhere explaining the difference between the 2 versions while you are at it.